### PR TITLE
new node api nick

### PR DESCRIPTION
**Pull request by youngsun4786**
INCLUDE `app.jpeg` and `output.jpeg` (DONE)
AI/ML NODES
- CLASSIFICATION/ACCURACY
- [x] app.txt 
- [x] example.md
- CLASSIFICATION/ONEHOTECODING
- [x] app.txt 
- [x] example.md
- CLASSIFICATION/SVM
- [x] app.txt 
- [x] example.md
- CLASSIFICATION/TRAIN_TEST_SPLIT
- [x] app.txt 
- [x] example.md
- IMAGE_CAPTIONING/NLP_TO_GPT2
- [x] app.txt 
- [x] example.md
- NLP/COUNT_VECTORIZER 
- [x] app.txt 
- [x] example.md
- PREDICT_TIME_SERIES/PROHPET_PREDICT
- [x] app.txt 
- [x] example.md
- REGRESSION/LEAST_SQUARES 
- [x] app.txt 
- [x] example.md
-  SEGMENTATION/DEEPLAB_V3 
- [x] app.txt 
- [x] example.md
- TEXT_SUMMARIZATION/BART_LARGE_CNN
- [x] app.txt 
- [x] example.md

EXTRACTOR NODES
 DONE

GENERATORS
- SAMPLE_DATASET/TEXT_DATASET

VISUALISERS
- PLOTLY/PROPHET_LINE
- [x] app.txt 
- [x] example.md
- PLOTLY/PROPHET_COMPONENT
- [x] app.txt 
- [x] example.md

### Description

- [x] I've included a concise description of what each node does

### Styleguide

- [x] My node adheres to the [styleguide](https://github.com/flojoy-io/nodes/blob/main/node_styleguide.md) for Flojoy nodes

### Docs

- [x] I've submitted a PR for a documentation page for the new node(s) that contains usage examples (see docs.flojoy.io)

### Testing

- [x] This PR includes a unit test ([example here](https://github.com/flojoy-io/nodes/blob/c28808f2d0aec8f116cdb3fccb46c4f5256574e9/TRANSFORMERS/ARITHMETIC/ADD/ADD_test_.py) and/or ideally a screenshot of the node's output on an example app.

### CODE REVIEW

File: AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py
Comment (JeffDotPng) [inline AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:25]: Use `isinstance`
Also if `b` can be None mark it as optional.

**Review by JeffDotPng — CHANGES_REQUESTED** [review]

File: VISUALIZERS/PLOTLY/PROPHET_PLOT/PROPHET_PLOT.py
Comment (youngsun4786) [reply VISUALIZERS/PLOTLY/PROPHET_PLOT/PROPHET_PLOT.py:7]: @smahmed776 said that `extra` key might be required for types other than `DataFrame` so I've made modifications accordingly in `data_container.py`. In that case, should I revert the changes in modify this node that it takes in two different inputs?

File: AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py
Comment (youngsun4786) [reply AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py:25]: I replaced the names of the token with an arange as like a temporary setting. Was going to create a PR for it but I'll make it return two DataContainers.

File: AI_ML/CLASSIFICATION/ACCURACY/output.jpeg
Comment (smahmed776) [inline AI_ML/CLASSIFICATION/ACCURACY/output.jpeg:1]: Both 'app.jpeg' and 'output.jpeg' should ideally be identical, with the only distinction being the visual nodes.

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (smahmed776) [inline AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py:4]: In my opinion, sklearn package should be a node dependency.

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (smahmed776) [inline AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py:13]: we should probably use all Literal values from scikit package for kernel parameter like: `kernel : {'linear', 'poly', 'rbf', 'sigmoid', 'precomputed'}` 
Also 'rvm' seems to be incorrect as this is not listed in SVC class docstring.

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (smahmed776) [inline AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py:5]: Consider removing unused imports

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/app.txt
Comment (smahmed776) [inline AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/app.txt:87]: Please use a relative path so that it works on any machine. e.x `PYTHON/nodes/EXTRACTORS/FILE/READ_CSV/assets/iris_train.csv`

File: AI_ML/IMAGE_CAPTIONING/NLP_CONNECT_VIT_GPT2/app.txt
Comment (smahmed776) [inline AI_ML/IMAGE_CAPTIONING/NLP_CONNECT_VIT_GPT2/app.txt:20]: Please use relative path so that it works on other machines

File: AI_ML/NLP/COUNT_VECTORIZER/app.txt
Comment (smahmed776) [inline AI_ML/NLP/COUNT_VECTORIZER/app.txt:18]: Same here use relative path

File: AI_ML/OBJECT_DETECTION/OBJECT_DETECTION/app.txt
Comment (smahmed776) [inline AI_ML/OBJECT_DETECTION/OBJECT_DETECTION/app.txt:20]: Again, use relative path

File: AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py
Comment (Roulbac) [inline AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:9]: Why not just support only dataframes?
I feel supporting "categories" and/or "columns" makes thise node more complicated and error prone.

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py:13]: I would even include a link to the SKLearn doc for SVM such that users understand what 'poly' or 'rbf' means.

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (Roulbac) [inline AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py:42]: How about supplying `training_labels` in the input as a `DataFrame | Matrix` such that the distinction is clear? Otherwise we have to make assumptions and this is prone to errors and misunderstanding

File: AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py
Comment (Roulbac) [inline AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:28]: Is this a typo?

**Review by Roulbac — COMMENTED** [review]
Few minor comments, let me know what you think.

Additionally, some node tests need to be updated to reflect the new node API. Do we have CI running these tests?

File: AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py
Comment (youngsun4786) [reply AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:28]: It was a check to cover cases when it was receiving inputs directly from linspace or not

File: AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py
Comment (youngsun4786) [reply AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:9]: That's a good point. I was actually running into errors when handling categories. However, I think columns params should be supported so that users can freely select any columns from data to be turned into OHE.

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (youngsun4786) [reply AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py:13]: @Roulbac @smahmed776 Do I have to follow the docs convention which uses `rbf` as default or can I stick to `linear` as the default?

**Comment by youngsun4786** [comment]
> Few minor comments, let me know what you think.
> 
> Additionally, some node tests need to be updated to reflect the new node API. Do we have CI running these tests?

yes we have a PyTest CI that runs all test files. I'll prioritize fixing up, perfecting the nodes and their corresponding `app.txt, example.md, app.jpeg, output.jpeg` and take a look at those if I must after nodes get merged!

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (youngsun4786) [reply AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py:42]: In that case, would it be better to separate the training data into training_label and training_features (extracted directly inside SVM node) and extract training label using `EXTRACT_COLUMNS` node?

File: AI_ML/CLASSIFICATION/ACCURACY/ACCURACY.py
Comment (JeffDotPng) [inline AI_ML/CLASSIFICATION/ACCURACY/ACCURACY.py:38]: same here

File: AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py
Comment (JeffDotPng) [inline AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py:34]: This is fine for now, but we should change the matrix return to a vector (vector type exists in frontier, I guess I forgot to add it to the actual flojoy package)

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py]: It's ok to stick with `linear`, as long as studio users can read the doc and understand what the other possible values mean

File: AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:9]: I agree 100% and it's ok to have that in one node unless you deem it unnecessary to have that "column selection" functionality be its own separate node (i.e. for users to select the columns they need, they can compose an "EXTRACT_COLUMN" node + "ONE_HOT_ENCODING" node). Please feel free to proceed with whichever option.

File: AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py]: Great idea. This would be modular, flexible and reusable. It would also make this `SVM` node simpler.

**Comment by Roulbac** [comment]
@youngsun4786  I replied to your comments about SVM + ONE_HOT_ENCODING. I think composability with other nodes for column extraction makes sense and applies the same way to both those nodes. Feel free to proceed, and I should be able to approve at the next round

File: AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py
Comment (JeffDotPng) [reply AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:28]: This just always overwrites the x values with the y values, giving the same line each time, so it should be deleted
Either way, we should probably have Linspace return a Vector instead of an OrderedPair, since having `None` as the x value is a bit misleading.
This should actually be `if not a.x` though.

**Review by JeffDotPng — CHANGES_REQUESTED** [review]

File: AI_ML/CLASSIFICATION/ACCURACY/ACCURACY.py
Comment (youngsun4786) [reply AI_ML/CLASSIFICATION/ACCURACY/ACCURACY.py:33]: woops my bad on that

File: AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py
Comment (youngsun4786) [reply AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py:34]: nice when you do add I'll update corresponding changes

File: AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py
Comment (youngsun4786) [reply AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:9]: That's also valid, in that case then I would need a second input which contains the extracted columns on which OHE can be performed. However, I just find it a bit uncomfortable for users, if they end up selecting all columns to yield OHE, then that process might be cumbersome.

Comment (youngsun4786) [reply AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:9]: Therefore, I kept the columns option

File: AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py
Comment (youngsun4786) [reply AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:28]: so do I just keep what's in the `else` statement?

Comment (JeffDotPng) [reply AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:28]: Yes, and you can delete the if condition (I just opened a PR to add a vector type so you don't need to handle this case anymore)

File: AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py
Comment (youngsun4786) [reply AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:9]: I made the changes!

**Review by Roulbac — APPROVED** [review]
Approving pending other folk's comments and requests for changes

File: AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py
Comment (JeffDotPng) [inline AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER.py:9]: Change the type here too

File: AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER_test_.py
Comment (JeffDotPng) [inline AI_ML/NLP/COUNT_VECTORIZER/COUNT_VECTORIZER_test_.py:45]: This test has to be updated to reflect the new multiple output format of the node

File: AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py
Comment (JeffDotPng) [inline AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:8]: This node should not take in vector, it doesn't make sense to compute a regression for a single array
I assume you're doing this because you're testing with linspace, but computing a regression for a straight line is kinda pointless

File: AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py
Comment (JeffDotPng) [inline AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py:9]: Returning vector doesn't make sense either, a regression has to have x and y coordinates

**Review by JeffDotPng — CHANGES_REQUESTED** [review]

**Pull request merged**
